### ISSUE TITLE
Should CLI tool wrapper scripts enforce locale env variables expected by Elixir?

### ISSUE THREAD

**Issue by michaelklishin** [rabbitmq/rabbitmq-server#1573]
See https://github.com/rabbitmq/rabbitmq-server/issues/1572 and https://github.com/elixir-lang/elixir/issues/3548.

TL;DR: Elixir unconditionally emits a warning (as of https://github.com/elixir-lang/elixir/commit/3ec0d2062171754eb4409dd5714dd744b1d806c3) when running in an environment where locale variables are set to a non-UTF-8 value. The warning can trip up tools such as Puppet that parse CLI tool output (we will not debate whether that's a good idea here).

While I don't buy into the argument that

> Forcing the env to a specific locale seems a bit drastic

we can easily force it in a way that Elixir requires in our wrapper scripts.

@dumbbell @hairyhum I recall we had this discussion before but the conclusion was never explicitly stated publicly.

Should we force locale/language variables for CLI tool invocations or does the potential downsides outweigh the gains for provisioning tools that have to parse CLI output?

Since we don't localise CLI tool output, I can't think of any potential confusing/breaking behaviour.

**Comment by michaelklishin** [comment]
Filed it here and not in https://github.com/rabbitmq/rabbitmq-server/issues/1573 because the wrappers are in this repo at the moment.

**Comment by wyardley** [comment]
Sorry, what I meant is (in terms of it being drastic), is just concern about causing an unintended consequence to the user when we start mucking around with default env in modules. For Puppet, at least, it should be only in the calls related to the module, but still I worried that it could cause unpredictable behavior (not to mention the political issue of what specific locale would be the default).

Either way, it's good to have confirmation that setting it to en_US.UTF-8 shouldn't cause an issue -- I didn't know whether users could create resources for which this might cause a problem in certain corner cases.

**Comment by michaelklishin** [comment]
We understand that there will be tension about whether this is a good idea and what default should be used. IIRC this is exactly why we didn't do it before. Requiring every user to configure locale settings would also suck.

FWIW our CLI tools are currently not localised and it's unlikely to change in the medium term, so there's less potential for breakage.

**Comment by michaelklishin** [comment]
@gerhard might have an opinion on this :)

**Comment by michaelklishin** [comment]
Closing in favor of https://github.com/rabbitmq/rabbitmq-website/issues/512. We'll update the Chef cookbook as well. Thank you for bringing this up, @wyardley.

**Comment by gerhard** [comment]
Thank you for the ping @michaelklishin, great resolution, fully agree.
